sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/Coverage.java

```diff
@@ -132,4 +112,21 @@ private void mergeLineHits(Coverage otherCoverage){
       }
     }
   }
+
+  private static BranchCoverage getBranchCoverage(int lineNumber, List<BranchPoint> branchPoints) {
+    // Mapping BranchPoints from different coverage reports is fragile due to potential differences in compilation.
+    // Therefore, we use a forgiving approach: First, count the number of BranchPoints per report and take the maximum. Then, map
+    // BranchPoints and count the ones that are covered. If the mapping did not go well, we might find too many covered BranchPoints.
+    // In this case, we cap the amount to the previously calculated maximum and assume all branches are covered.
+    int maxBranchPoints = branchPoints.stream()
```

Comment:
Can we store the stream for reuse in both `maxBranchPoints` and `coveredBranchPoints`?

Reply:
This is the beauty of Java streams! You cannot reuse them once they have been collected 😖

Reply:
I see, all good then!